### PR TITLE
[server] Add server API to list blocked repositories

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Add `adminGetBlockedRepositories` API to `components/server` to fetch the list of blocked repositories.

This is a step towards using these endpoints to manage blocked repositories in the admin UI.

## Related Issue(s)

Part of #11030

## How to test

* Follow the first 3 steps on https://github.com/gitpod-io/gitpod/pull/11036 to add some blocked repositories.

* Open the developer tools on the admin page for this preview environment.

* Run:
```js
await window._gp.gitpodService.server.adminGetBlockedRepositories({offset: 0, limit: 10, orderBy: 'id', orderDir: "desc", searchTerm: "foo"})
```

setting `searchTerm` as appropriate (or omitting it entirely).

## Release Notes

```release-note
NONE
```

## Werft options

- [x] /werft with-preview

/hold

### CODE REVIEW

File: components/gitpod-db/src/blocked-repository-db.ts
Comment (geropl) [inline components/gitpod-db/src/blocked-repository-db.ts:15]: Here it might make sense to limit to the fields that actually are indexex, for instance by using `orderBy: keyof Pick<BlockedRepository, "id | "blockUser">` or similar.

File: components/gitpod-db/src/typeorm/blocked-repository-db-impl.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/blocked-repository-db-impl.ts:39]: Note that this won't work: `andWhere` strictly requires a prior `where` this is a typeorm oddity... :see_no_evil: 
Usually it can be solved by having something general as first clause, maybe `.where("br.deleted = FALSE")`.

**Comment by geropl** [comment]
Left some comments. Please squash before merging! :upside_down_face:

File: components/gitpod-db/src/blocked-repository-db.ts
Comment (andrew-farries) [reply components/gitpod-db/src/blocked-repository-db.ts:15]: We have no indexes on the table besides the PK. As mentioned yesterday https://github.com/gitpod-io/gitpod/pull/11036#discussion_r910989383, mysql can't make use of indexes on `urlRegexp` when testing with `REGEXP`.

Do you think we should add indexes to support this admin search?

File: components/gitpod-db/src/typeorm/blocked-repository-db-impl.ts
Comment (geropl) [reply components/gitpod-db/src/typeorm/blocked-repository-db-impl.ts:39]: > This does seem to work though; tests using the searchTerm pass and it behaves as expected when tested manually too.

Uh, that's nice! Might have been introduced with the typeorm version bump we did last October - thx for sharing!

**Comment by geropl** [comment]
/werft run


:crossed_fingers:

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works! :heavy_check_mark: 

/hold Please squash before merging! :upside_down_face:

**Pull request merged**
